src/ripple/protocol/STObject.h

```diff
@@ -481,6 +481,13 @@ class STObject
     void setAccountID (SField const& field, AccountID const&);
 
     void setFieldAmount (SField const& field, STAmount const&);
+    template<class Integer,
+        class = typename std::enable_if_t <
+            std::is_integral<Integer>::value>>
+    void setFieldAmount (SField const& field, tagged_integer<Integer, DropsTag> const& v)
```

Comment:
I'm wondering if you really need this SFINAE (I've seen it in quite a few places that seem redundant), I think you can just use `template<typename Integer> void setFieldAmount(...);` Since you can't declare an invalid tagged_integer, it has all the necessary `static_assert`s already.